### ISSUE TITLE
How to provide identity as string instead of a file?

### ISSUE THREAD

**Issue by PatrickHuetter** [int128/gradle-ssh-plugin#183]
I'm using gitlab-ci for continuous integration of a grails 3 application. The grails application gets build within a docker container and deployed via gradle with the gradle-ssh-plugin to a linux-server.

I wouldn't like to provide the ssh key within my application or within the vcs repository. With gitlab-ci it's possible to set environment variables for each project which get injected on each build into the docker container. I would like to set the identity into an environment variable and read this variable within the gradle-ssh-configuration. Using a password this is already possible but using a key instead would be much better. So is it possible to set the identity by string instead of a file? If not, this would be a good feature.

At this time i'm reading the environment variable into a string variable and writing the string into a temporary file and then provide this file to the gradle ssh plugin.

**Comment by int128** [comment]
Now released 1.2.0. Identity can be provided as a File or String as follows:
```groovy
ssh.remotes {
  server {
    host = 'localhost'
    user = System.getProperty('user.name')
    identity = """-----BEGIN RSA PRIVATE KEY-----
xxxxx
-----END RSA PRIVATE KEY-----"""
  }
}
ssh.run {
  session(ssh.remotes.server) {
    execute 'id'
  }
}
```

**Comment by sivaboomamurugan** [comment]
SSH plugin was really a wonderful one, it saves a lot of time in my work.

I am facing a issue though in doing a ssh connection via private key, I have my private key in the ~/.ssh/id_rsa location and i thought mentioning the identity: file('~/.ssh/id_rsa') should work but unfortunately it resolving the file path relative to the current directory from where the build is executing.

build.gradle file is located at /home/user/myapp
when executing the task, it gives me a file not found error as it looking for the file in /home/user/myapp/~/.ssh/id_rsa

any idea what am i missing or this is the intended behavior
